File: src/main/kotlin/hanablive/model/dto/instruction/GameInitData.kt
package eelst.ilike.hanablive.model.dto.command

import eelst.ilike.game.PlayerId
import eelst.ilike.hanablive.model.dto.GameOptions
import java.time.ZonedDateTime

data class GameInitData(
    val tableID: Int,
    val playerNames: List<PlayerId>,
    val ourPlayerIndex: Int,
    val spectating: Boolean,
    val shadowing: Boolean,
    val replay: Boolean,
    val databaseID: Int,
    val hasCustomSeed: Boolean,
    val seed: String,
    val datetimeStarted: ZonedDateTime,
    val datetimeFinished: ZonedDateTime,
    val options: GameOptions,
    val characterAssignments: List<Any>,
    val characterMetadata: List<Any>,
    val sharedReplay: Boolean,
    val sharedReplayLeader: PlayerId,
    val sharedReplaySegment: Int,
    val sharedReplayEffMod: Int,
    val paused: Boolean,
    val pausePlayerIndex: Int,
    val pauseQueued: Boolean
)

File: src/external/hanablive/dto/game_init_data.rs
use chrono::{DateTime, Utc};
use serde::Deserialize;

use super::game_options::GameOptions;

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct GameInitData {
    #[serde(rename = "tableID")]
    pub table_id: usize,
    pub player_names: Vec<String>,
    pub our_player_index: usize,
    pub spectating: bool,
    pub shadowing: bool,
    pub replay: bool,
    #[serde(rename = "databaseID")]
    pub database_id: i32,
    pub has_custom_seed: bool,
    pub seed: String,
    pub datetime_started: DateTime<Utc>,
    pub datetime_finished: DateTime<Utc>,
    pub options: GameOptions,
    pub character_assignments: Vec<serde_json::Value>,
    pub character_metadata: Vec<serde_json::Value>,
    pub shared_replay: bool,
    pub shared_replay_leader: String,
    pub shared_replay_segment: usize,
    pub shared_replay_eff_mod: usize,
    pub paused: bool,
    pub pause_player_index: i32,
    pub pause_queued: bool,
}
